plutus-core/plutus-core/src/PlutusCore/Core/Instance/Pretty/Readable.hs

```diff
@@ -18,11 +16,24 @@ import PlutusCore.Core.Type
 import PlutusCore.Pretty.PrettyConst
 import PlutusCore.Pretty.Readable
 
-import Data.Void
 import Prettyprinter
 import Prettyprinter.Custom
 import Universe
 
+-- | Split an iterated 'KindArrow' (if any) into the list of argument types and the resulting type.
+viewKindArrow :: Kind ann -> Maybe ([Kind ann], Kind ann)
+viewKindArrow kind0@KindArrow{} = Just $ go kind0 where
+    go (KindArrow _ dom cod) = first (dom :) $ go cod
+    go kind                  = ([], kind)
+viewKindArrow _ = Nothing
+
+-- | Split an iterated 'TyFun' (if any) into the list of argument types and the resulting type.
+viewTyFun :: Type tyname uni ann -> Maybe ([Type tyname uni ann], Type tyname uni ann)
+viewTyFun ty0@TyFun{} = Just $ go ty0 where
+    go (TyFun _ dom cod) = first (dom :) $ go cod
+    go ty                = ([], ty)
+viewTyFun _ = Nothing
+
```

Comment:
Sorry for the late review, I just remembered about these:

https://github.com/input-output-hk/plutus/blob/a1b9bb57d5342d7e757b4d8de3c70f456b425a92/plutus-core/plutus-ir/src/PlutusIR/Compiler/Datatype.hs#L67-L85

Can we merge them together?